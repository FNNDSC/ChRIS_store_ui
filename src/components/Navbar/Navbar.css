header {
  /* shadow */
  box-shadow: 0 20px 20px -20px rgba(0, 0, 0, 0.5);
}

nav.navbar-default {
  /* display */
  height: 132px;

  /* position */
  position: relative;
}

div.navbar-row {
  /* display */
  display: flex;
}

div.navbar-row > .navbar-header {
  /* flex */
  flex: 1;
}

header .row > .navbar-header > .navbar-logo {
  /* display */
  padding: 0 !important;
  margin: 8px 0;

  /* other */
  outline: 0;
}

.navbar-logo-img {
  /* display */
  width: 243px;
  height: 48px;
}

.navbar-header > .navbar-search.search-pf {
  /* color */
  color: var(--pf-black-200);
  
  /* display */
  display: block;
  
  /* position */
  position: absolute;
  top: 66px;
  left: 0;
  right: 0;
}

.navbar-header > .navbar-search.search-pf > .form-group {
  /* display */
  display: block;
}

.navbar-search > .has-icon .search-icon {
  /* display */
  display: flex;
  width: 64px;
  height: 64px;

  /* font */
  font-size: 1.5em;

  /* flex */
  flex-direction: column;
  justify-content: center;
}

.navbar-search .form-group .form-control {
  /* color */
  color: var(--pf-black-200);
  background: var(--pf-black-900);

  /* display */
  padding: 32px 66px;

  /* font */
  font-size: 1.2em;
  font-weight: 600;

  /* border */
  border: 0;
  border-radius: 0;

  /* other */
  box-shadow: none;
}

.navbar-search .form-control::placeholder {
  /* color */
  color: var(--pf-black-200);

  /* font */
  font-style: normal;
}

.navbar-trigger {
  /* color */
  color: var(--pf-black-200);

  /* display */
  margin: 12px 15px;
  padding: 0 10px;

  /* font */
  font-size: 2em;
  
  /* other */
  float: right;
  cursor: pointer;
}

/* ===== INACTIVE NAV LINKS ===== */

header .navbar.navbar-default .navbar-right > li > a:hover {
  /* color */
  color: var(--pf-black-300);
}

header .navbar.navbar-default .navbar-right > li > a {
  /* color */
  color: var(--pf-black-100);

  /* display */
  padding: 22px 15px;

  /* font */
  font-size: 1.2em;
}

header .navbar.navbar-default .navbar-right > li > button {
  /* display */
  margin: 16px 13px;
}

/* ====== ACTIVE NAV LINKS ====== */

header .navbar.navbar-default .navbar-right > li > a.active {
  /* color */
  color: var(--pf-black-100);

  /* other */
  cursor: default;
}

header .navbar.navbar-default .navbar-right > li > a.active:hover {
  /* color */
  background: inherit;
}

/* ======= NAVBAR DROPDOWN ====== */

header > .navbar-dropdown {
  /* color */
  background: #1d1d1d;

  /* border */
  border-bottom: 1px solid var(--pf-black-800);

  /* other */
  z-index: 1;
}

.navbar-btn-container {
  /* font */
  line-height: 1;
}

/* ==== INACTIVE DROPDOWN BTN === */

.navbar-dropdown-btn {
  /* display */
  display: block;
  padding: 0.5em;
  
  /* color */
  color: var(--pf-black-100);

  /* font */
  font-size: 2em;
  font-weight: 600;
}

.navbar-dropdown-btn:hover {
  /* color */
  background: var(--pf-black-900);
}

/* ===== ACTIVE DROPDOWN BTN ==== */

.navbar-dropdown-btn.active {
  /* color */
  background: var(--pf-black);
  color: var(--pf-black-100);

  /* other */
  cursor: default;
}

.navbar-btn-container .navbar-dropdown-btn:hover,
.navbar-btn-container .navbar-dropdown-btn:focus {
  color: var(--pf-black-100);
  text-decoration: none;
}

/* ========= RESPONSIVE ========= */

@media (min-width: 480px) {
  header .navbar-header > .navbar-logo {
    /* display */
    margin: 0 0 0 25px;
  }
}

@media (min-width: 768px) {
  nav.navbar-default { height: 66px; }

  .navbar-header > .navbar-search.search-pf { 
    /* color */
    color: inherit;

    /* display */
    display: table;
    margin: 19px 0;
    padding-left: 1em;
    
    /* position */
    position: static; 
  }

  .navbar-search .form-group {
    /* position */
    position: relative;

    /* display */
    margin: 0;
  }

  .navbar-search > .has-icon .search-icon {
    /* color */
    color: var(--pf-black-300);

    /* display */
    width: 1.35em;
    height: 1.35em;
  
    /* font */
    font-size: 1.5em;
  }

  .navbar-search .form-group .form-control {
    /* color */
    background: inherit;
    color: var(--pf-black-300);

    /* display */
    padding: 0 0 0 2em;

    /* font */
    font-size: 1.3em;
    font-weight: 600;
  }

  .navbar-search .form-control::placeholder {
    /* color */
    color: var(--pf-black-300);
  }

  .navbar-trigger, .navbar-dropdown { display: none; }
  .collapse.in { display: none !important; }
}

@media (min-width: 1110px) {
  header .row > .navbar-header > .navbar-logo {
    /* display */
    margin-left: 0;
  }
}
