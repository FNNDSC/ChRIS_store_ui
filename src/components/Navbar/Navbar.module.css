.login-button .other-button{
  background: #06c !important;
  color: white !important;
}