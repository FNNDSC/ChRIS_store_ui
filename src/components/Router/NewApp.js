import React from 'react';

const NewApp = () => (
  <div> New App </div>
);

export default NewApp;
