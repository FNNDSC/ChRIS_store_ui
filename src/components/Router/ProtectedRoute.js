import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import ChrisStore from '../../store/ChrisStore';

const ProtectedRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) => {
      const { store } = rest;
      const isLoggedIn = store.get('isLoggedIn');
      if (isLoggedIn) {
        return <Component {...props} />;
      }
        return <Redirect to={{
          pathname: "/signin",
          state: {from: props.location},
        }} />;
    }}
  />
);


ProtectedRoute.propTypes = {
  store: PropTypes.objectOf(PropTypes.object),
  component: PropTypes.oneOfType([PropTypes.func]).isRequired,
};
ProtectedRoute.defaultProps = {
  store: {},
};

export default ChrisStore.withStore(ProtectedRoute);
