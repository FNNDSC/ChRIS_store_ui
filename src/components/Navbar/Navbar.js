import React, { useState } from 'react';
import {
  Brand,
  PageHeader,
  Nav,
  NavList,
  NavItem,
  PageHeaderTools,
  PageHeaderToolsItem,
} from '@patternfly/react-core';
import Button from '../Button';
import { NavLink, useHistory } from 'react-router-dom';
import Search from './components/Search/Search';
import './Navbar.css';
import LogoImg from '../../assets/img/chris-plugin-store_logo.png';
import ChrisStore from '../../store/ChrisStore';

const navLinks = [
  {
    label: 'Plugins',
    to: '/plugins'
  },
  {
    label: 'Quick Start',
    to: '/quickstart'
  },
  {
    label: 'Dashboard',
    to: '/dashboard',
    cond: (store) => store.get('isLoggedIn')
  }
];

/**
 * Conditionally renders a list of links into a <Nav>.
 */
const Navigation = ({ store }) => {
  const [searchKey, setSearchKey] = useState('');
  const history = useHistory();
  const shouldShowLink = (linkInfo) => {
    if (!linkInfo.cond) {
      return true;
    }
    return linkInfo.cond(store);
  };
  return (
    <Nav variant="horizontal">
      <NavList>
        {
          navLinks
            .filter((l) => shouldShowLink(l))
            .map(link => (
              <NavItem
                key={link.to}
                itemId={link.to}
                isActive={window && window.location.pathname === link.to}>
                <NavLink to={link.to} activeClassName="pf-m-current">
                  {link.label}
                </NavLink>
              </NavItem>
            ))
        }
        <NavItem>
          <Search 
            placeholder="Search Plugin"
            value={searchKey}
            onChange={(value) => setSearchKey(value)}
            onClear={() => setSearchKey('')}
            onSearch={() => {history.push('/plugins')}}
          />
        </NavItem>
      </NavList>
    </Nav>
  );
}

const StatefulNavigation = ChrisStore.withStore(Navigation);
const statefulNavigation = (<StatefulNavigation />);

const LoginButton = ({ store }) => (
  <NavLink to="/signin">
    <Button
      variant="primary"
      className="login-button"
    >
      {store.get('isLoggedIn') ? 'Sign Out' : 'Sign In'}
    </Button>
  </NavLink>
);
const StatefulLoginButton = ChrisStore.withStore(LoginButton);

const HeaderTools = (
  <PageHeaderTools>
    <PageHeaderToolsItem>
      <StatefulLoginButton />
    </PageHeaderToolsItem>
  </PageHeaderTools>
)

const Logo = (<Brand alt="ChRIS Plugin Store" src={LogoImg}/>);

const Navbar = () => (
  <PageHeader
    logo={Logo}
    logoComponent={NavLink}
    logoProps={{to: '/'}}
    topNav={statefulNavigation}
    headerTools={HeaderTools}
  />
);

export default Navbar;
