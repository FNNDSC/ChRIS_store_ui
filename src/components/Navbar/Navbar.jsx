import React from 'react';
import {
  Button,
  Brand,
  PageHeader,
  Nav,
  NavList,
  NavItem,
  PageHeaderTools,
  PageHeaderToolsItem,
} from '@patternfly/react-core';
import { NavLink } from 'react-router-dom';
import './Navbar.css';
import LogoImg from '../../assets/img/chris-plugin-store_logo.png';
import ChrisStore from '../../store/ChrisStore';

const navLinks = [
  {
    label: 'Plugins',
    to: '/plugins',
  },
  {
    label: 'Pipelines',
    to: '/pipelines',
  },
  {
    label: 'Submit your Plugin',
    to: '/quickstart',
  },
  
  {
    label: 'Dashboard',
    to: '/dashboard',
    if: (store) => store.get('isLoggedIn')
  }
];

/**
 * Conditionally renders a list of links into a <Nav>.
 */
const Navigation = ChrisStore.withStore(({ store }) => (
  <Nav variant="horizontal">
    <NavList>
      {
        navLinks
          .filter(link => {
            if (!link.if)
              return true;
            return link.if(store);
          })
          .map(link => (
            <NavItem
              key={link.to}
              itemId={link.to}
              isActive={window && window.location.pathname === link.to}>
              <NavLink to={link.to} activeClassName="pf-m-current">
                {link.label}
              </NavLink>
            </NavItem>
          ))
      }
    </NavList>
  </Nav>
));

const Navbar = ({ searchComponent, store }) => {
  const HeaderTools = (
    <PageHeaderTools>
      <PageHeaderToolsItem>
        {searchComponent}
      </PageHeaderToolsItem>
      <PageHeaderToolsItem>
        <NavLink to="/signin">
          <Button id="login-button">
            {store.get('isLoggedIn') ? 'Sign Out' : 'Sign In'}
          </Button>
        </NavLink>
      </PageHeaderToolsItem>
    </PageHeaderTools>
  )

  const Logo = (<Brand className="logo" alt="ChRIS Plugin Store" src={LogoImg}/>)
  
  return (
    <PageHeader
      logo={Logo}
      logoComponent={NavLink}
      logoProps={{to: '/'}}
      topNav={(<Navigation />)}
      headerTools={HeaderTools}
    />
  )
}
export default ChrisStore.withStore(Navbar);
