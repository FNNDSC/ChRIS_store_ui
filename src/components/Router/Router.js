import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Navbar from '../Navbar/Navbar';
import Footer from '../Footer/Footer';

// pages
import Welcome from '../Welcome/Welcome';
import Plugins from '../Plugins/Plugins';
import Plugin from '../Plugin/Plugin';
import Developers from '../Developers/Developers';
import CreatePlugin from '../CreatePlugin/CreatePlugin';

const Router = () => (
  <div>
    <Navbar />
    <Switch>
      <Route exact path="/" component={Welcome} />
      <Route path="/plugins" component={Plugins} />
      <Route path="/plugin/:plugin" component={Plugin} />
      <Route path="/developers" component={Developers} />
      <Route path="/create" component={CreatePlugin} />
    </Switch>
    <Footer />
  </div>
);

export default Router;
