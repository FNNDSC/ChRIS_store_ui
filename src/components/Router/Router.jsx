import React from 'react';
import { Route, Switch } from 'react-router-dom';
import AppLayout from '../AppLayout/AppLayout';

// pages
import Welcome from '../Welcome/Welcome';
import PluginDetail from '../Plugin/Plugin';
import PluginMetaDetail from '../Plugin/PluginMeta';
import PluginsList from '../Plugins/Plugins';
import Developers from '../Developers/Developers';
import CreatePlugin from '../CreatePlugin/CreatePlugin';
import NotFound from '../NotFound/NotFound';
import Dashboard from '../Dashboard/Dashboard';
import DashCollaboratorView from '../Dashboard/components/DashCollaborator/DashCollaboratorView';
import PipelineCatalog from '../Pipelines/PipelineCatalog';
import ProtectedRoute from './ProtectedRoute';

const Router = () => (
  <AppLayout>
    <Switch>
      <Route path="/plugin/:pluginName" component={PluginMetaDetail} />
      <Route path="/p/:pluginId" component={PluginDetail} />
      <Route path="/plugins" component={PluginsList} />
       <Route path="/pipelines" component={PipelineCatalog} />
      
      <ProtectedRoute path="/create" component={CreatePlugin} />
      <ProtectedRoute path="/dashboard" component={Dashboard} />
      <ProtectedRoute path="/manage/collaborators/:pluginName" component={DashCollaboratorView} />

      <Route path="/quickstart" component={Developers} />
      <Route exact path="/" component={Welcome} />
      <Route component={NotFound} />
    </Switch>
  </AppLayout>
);

export default Router;
