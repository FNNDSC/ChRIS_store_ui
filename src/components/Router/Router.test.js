import React from 'react';
import { shallow } from 'enzyme';
import Router from './Router';
import Navbar from '../Navbar/Navbar';

describe('Router', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Router />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render BrowserRouter component', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
  });

  it('should render div element', () => {
    expect(wrapper.find('div')).toHaveLength(1);
  });

  it('should render Navbar component', () => {
    expect(wrapper.containsMatchingElement(<Navbar />)).toEqual(true);
  });

  it('should render a Switch coponent', () => {
    expect(wrapper.find('Switch')).toHaveLength(1);
  });

  it('should render 4 Routes', () => {
    expect(wrapper.find('Route')).toHaveLength(4);
  });

  it('every route should have a path prop and component prop', () => {
    const routes = wrapper.find('Route');
    routes.forEach((route) => {
      expect(route.prop('path')).toBeDefined();
      expect(route.prop('component')).toBeDefined();
    });
  });

  it('should render a Footer component', () => {
    expect(wrapper.find('Footer')).toHaveLength(1);
  });
});
