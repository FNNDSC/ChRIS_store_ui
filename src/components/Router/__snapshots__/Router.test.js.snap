// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Router />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar
          location={
                    Object {
                              "search": "",
                            }
          }
/>,
        <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/plugins"
          />
          <Route
                    component={[Function]}
                    path="/plugin/:plugin"
          />
          <Route
                    component={[Function]}
                    path="/developers"
          />
          <Route
                    component={[Function]}
                    path="/create"
          />
</Switch>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": Object {
            "search": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/plugins"
/>,
            <Route
              component={[Function]}
              path="/plugin/:plugin"
/>,
            <Route
              component={[Function]}
              path="/developers"
/>,
            <Route
              component={[Function]}
              path="/create"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/plugins",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/plugin/:plugin",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/developers",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/create",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar
            location={
                        Object {
                                    "search": "",
                                  }
            }
/>,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/plugins"
            />
            <Route
                        component={[Function]}
                        path="/plugin/:plugin"
            />
            <Route
                        component={[Function]}
                        path="/developers"
            />
            <Route
                        component={[Function]}
                        path="/create"
            />
</Switch>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": Object {
              "search": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/plugins"
/>,
              <Route
                component={[Function]}
                path="/plugin/:plugin"
/>,
              <Route
                component={[Function]}
                path="/developers"
/>,
              <Route
                component={[Function]}
                path="/create"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/plugins",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/plugin/:plugin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/developers",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/create",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
